src/package-resolver.js

```diff
@@ -477,12 +475,12 @@ export default class PackageResolver {
       return;
     }
 
-    const fetchKey = `${req.registry}:${req.pattern}`;
+    const fetchKey = `${req.registry}:${req.pattern}:${String(req.optional)}`;
 
-    if (this.fetchingPatterns[fetchKey]) {
+    if (this.fetchingPatterns.has(fetchKey)) {
       return;
     } else {
-      this.fetchingPatterns[fetchKey] = true;
+      this.fetchingPatterns.add(fetchKey);
```

Comment:
this else block isn't necessary since other wise it'll just return?

Reply:
That's right, I'll fix it.

Reply:
Cost me about 3 hours to catch it :D